### PR TITLE
SubModule AddedFinished

### PR DESCRIPTION
**Pull request by Kajanthuka**

### CODE REVIEW

File: defect-tracker-server/src/main/java/com/defect/tracker/util/EndpointURI.java [hunk 1]
```diff
@@ -47,6 +47,7 @@
 	public static final String getSubModule = BASE_API_PATH + "SubModule/{moduleId}";
 	public static final String UpdateSubModule = BASE_API_PATH + "SubModule";
 	public static final String SUBMODULE_DELETE = BASE_API_PATH + "deleteSubModule/{id}";
+	public static final String SUBMODULE_ADD = BASE_API_PATH + "SubModuleAdd";
```
Comment (ramz7) [inline defect-tracker-server/src/main/java/com/defect/tracker/util/EndpointURI.java]: give submodule

File: defect-tracker-server/src/main/resources/application-dev.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
-server.port=1724
+server.port=1725
```
Comment (ramz7) [inline defect-tracker-server/src/main/resources/application-dev.properties:1]: no need to add application-dev properties file

File: defect-tracker-server/src/main/java/com/defect/tracker/controller/SubModuleController.java [hunk 3]
```diff
@@ -63,5 +64,18 @@
 
 	
 	
+	@PostMapping(value = EndpointURI.SUBMODULE_ADD)
+	public ResponseEntity<Object> createSubModule(@RequestBody SubModuleDto subModuleDto){
+		if(subModuleService.isSubModuleExistsByName(subModuleDto.getSubModuleName())) {
```
Comment (ramz7) [inline defect-tracker-server/src/main/java/com/defect/tracker/controller/SubModuleController.java]: check submodule name validation to same module id

File: defect-tracker-server/src/main/resources/application-dev.properties (hunk 2 above)
Comment (Kajanthuka) [reply defect-tracker-server/src/main/resources/application-dev.properties:1]: okay no add next time

File: defect-tracker-server/src/main/java/com/defect/tracker/controller/SubModuleController.java (hunk 3 above)
Comment (Kajanthuka) [reply defect-tracker-server/src/main/java/com/defect/tracker/controller/SubModuleController.java]: yes Progress ,

File: defect-tracker-server/src/main/java/com/defect/tracker/util/EndpointURI.java (hunk 1 above)
Comment (Kajanthuka) [reply defect-tracker-server/src/main/java/com/defect/tracker/util/EndpointURI.java]: Accept the changing

File: defect-tracker-server/src/main/java/com/defect/tracker/controller/SubModuleController.java (hunk 3 above)
Comment (Kajanthuka) [reply defect-tracker-server/src/main/java/com/defect/tracker/controller/SubModuleController.java]: I did It

**Pull request merged**